### ISSUE TITLE
No match between VMServiceScrape and Service for coreDNS, kubeScheduler and kubeControllerManager in latest victoria-metrics-k8s-stack chart

### ISSUE THREAD

**Issue by barthy1** [VictoriaMetrics/helm-charts#1282]
When I did update of victoria-metrics-k8s-stack  installation to the latest available chart version (0.25.2) I noticed that the targets for coreDNS, kubeScheduler and kubeControllerManager are 0/0 now.

After some investigation I figured out, that in the corresponding VMServiceScrape template there is hardcoded selector.matchLabels `app: {{ $fullname }}`
For instance for coreDNS
```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMServiceScrape
metadata:
  name: {{ $fullname }}
  namespace: {{ .Release.Namespace }}
  labels: {{ include "victoria-metrics-k8s-stack.labels" . | nindent 4 }}
spec:
  {{- omit .Values.coreDns.spec "selector" "namespaceSelector" | toYaml | nindent 2 }}
  selector:
    matchLabels:
      app: {{ $fullname }}
      app.kubernetes.io/instance: {{ default .Release.Name .Values.argocdReleaseOverride | trunc 63 | trimSuffix "-" }}
  namespaceSelector:
    matchNames:
      - "kube-system"
```

However in the corresponding Service same label is not generated. I see some code, that pretends to do it, but I can see in my actual deployment and even with `helm template` test that there is no label `app: {{ $fullname }}` in the service.
For example for coreDns template is

```
apiVersion: v1
kind: Service
metadata:
  name: {{ $fullname }}
  namespace: kube-system
  {{- $extraLabels := dict "app" $fullname "jobLabel" "coredns" }}
  {{- $ctx := merge (deepCopy $global) (dict "extraLabels" $extraLabels) }}
  labels: {{ include "victoria-metrics-k8s-stack.labels" $ctx | nindent 4 }}
spec:
  clusterIP: None
  ports:
    - name: http-metrics
      port: {{ .Values.coreDns.service.port }}
      protocol: TCP
      targetPort: {{ .Values.coreDns.service.targetPort }}
  {{- with .Values.coreDns.service.selector }}
  selector: {{ toYaml . | nindent 4 }}
  {{- end}}
{{- end }}
```
The output with `helm template`

```
# Source: victoria-metrics-k8s-stack/templates/servicemonitors/coredns.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-victoria-metrics-k8s-stack-coredns
  namespace: kube-system
  labels:
    helm.sh/chart: victoria-metrics-k8s-stack-0.25.0
    app.kubernetes.io/name: victoria-metrics-k8s-stack
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v1.102.1"
    app.kubernetes.io/managed-by: Helm
spec:
  clusterIP: None
  ports:
    - name: http-metrics
      port: 9153
      protocol: TCP
      targetPort: 9153
  selector:
    k8s-app: kube-dns
```

So, there is `app: {{ $fullname }}` in the spec.selector.matchLabels of VMServiceScrape, but no such label in the Service and it leads to the lack of metrics for coreDNS, kubeScheduler and kubeControllerManager

**Comment by AndrewChubatiuk** [comment]
this was an issue in 0.25.0, but was fixed in 0.25.1

**Comment by AndrewChubatiuk** [comment]
hey @barthy1. any updates?

**Comment by arteonprifti** [comment]
hello there @AndrewChubatiuk I don't know if it was related to this, but now the label `jobLabel` is missing for the kube-controller-manager service `jobLabel: kube-controller-manager` and now there is a `k8s-app: kube-controller-manager` label. Same thing for kube-proxy.
The job label is now: `job="victoria-metrics-k8s-stack-kube-controller-manager` and this causes the default alert to trigger
https://github.com/VictoriaMetrics/helm-charts/blob/90a767476a109e401d0cc4d2115f25c8ad4045f7/charts/victoria-metrics-k8s-stack/files/rules/generated/kubernetes-system-controller-manager.yaml#L10

**Comment by AndrewChubatiuk** [comment]
hey @arteonprifti 
[this](https://github.com/VictoriaMetrics/helm-charts/pull/1363) PR should address issues you're having

**Comment by AndrewChubatiuk** [comment]
@arteonprifti please check a new release it contains a fix

**Comment by arteonprifti** [comment]
Hi @AndrewChubatiuk sorry it took me some time to test this. I saw that your changes include now a `jobName` label which has the correct value, but it doesn't match with this: https://github.com/VictoriaMetrics/helm-charts/blob/046d071529ea3b39101e57e3876c307831150d28/charts/victoria-metrics-k8s-stack/values.yaml#L1056
Setting `jobLabel: jobName` does work, but wouldn't it be better to just set the `jobLabel` instead, since this affects the core-dns and kube-proxy as well?

**Comment by AndrewChubatiuk** [comment]
looks like this is a typo, will be fixed in 0.25.12

**Comment by arteonprifti** [comment]
yeap, now it matches, thank you very much
